debug/testlib.py

```diff
@@ -496,6 +496,29 @@ def wait_until_running(self, harts):
             if time.time() - start > self.timeout:
                 raise TestLibError("Timed out waiting for targets to run.")
 
+    def set_available(self, harts):
+        """Set the given harts to available, and any others to be
+        unavailable."""
+        available_mask = 0
+        for hart in harts:
+            available_mask |= 1 << hart.id
+        self.command(f"riscv dmi_write 0x1f 0x{available_mask:x}")
```

Comment:
So, 0x1F in this context isn't a JTAG IR value (bypass), it's a DMI address ("custom"), which Spike implements "unavailable" control.

Reply:
Yes. I should probably add a comment explaining that.